src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs

```diff
@@ -117,7 +117,7 @@ public bool HasListeners()
 
             ActivityDataRequest dataRequest = ActivityDataRequest.None;
 
-            if (parentId != null)
+            if (parentId != null && !Activity.TryConvertIdToContext(parentId, out context))
```

Comment:
I would suggest only converting the id as a fallback if the GetDataRequestByParentId callback wasn't provided. This lets people write optimized implementations of a sampler that skip the CPU and allocation costs of conversion if they want to. I am aiming that it should be possible for people to write optimized samplers that operate in double-digit ns. Based on benchmarking from the other day I'd expect this call alone to be at least 50ns, and probably a good bit more.

Reply:
@cijothomas are you ok with what @noahfalk suggesting here? I assume it should be ok but wanted to confirm before proceeding applying the suggestion.

Reply:
Yes. this sounds better.